lib/Conversion/TritonGPUToLLVM/ConvertLayoutOpToLLVM/SharedToDotOperandMFMA.cpp

```diff
@@ -220,6 +220,164 @@ Value computeBasePtr(ConversionPatternRewriter &rewriter, Location loc,
   return base;
 }
 
+/**
+ * @brief try find if value is an integer constant
+ * 
+ * Trace def-use chain and return integer in case we can proof it is constant.
+ * Current implementation can trace chains of insertValue->extractValue operations.
+ * 
+ * @param val Value for that we want to get constant
+ * @return std::optional on found integer value or empty std::optional
+*/
+std::optional<int> findConstValue(Value val) {
+  while (val && !val.getDefiningOp<LLVM::ConstantOp>()) {
+    LLVM::ExtractValueOp extractValOp = val.getDefiningOp<LLVM::ExtractValueOp>();
+    if (!extractValOp)
+      return std::optional<int>();
+    auto extractPosArr = extractValOp.getPosition();
+    if (extractPosArr.size() > 1)
+      return std::optional<int>();
+    int extractPos = extractPosArr[0];
+
+    int insertPos = -1;
+    LLVM::InsertValueOp insertValOp;
+    Value container = extractValOp.getOperand();
+    do {
+      insertValOp = container.getDefiningOp<LLVM::InsertValueOp>();
+      if (!insertValOp)
+        return std::optional<int>();
+      auto insertPosArr = insertValOp.getPosition();
+      if (insertPosArr.size() > 1)
+        return std::optional<int>();
+      insertPos = insertPosArr[0];
+      container = insertValOp.getContainer();
+    } while(insertPos != extractPos);
+    val = insertValOp.getValue();
+  }
+  if (!val)
+    return std::optional<int>();
+  auto cOp = val.getDefiningOp<LLVM::ConstantOp>();
+  assert(cOp);
+  auto valAttr = cOp.getValueAttr();
+  auto intAttr = dyn_cast<mlir::IntegerAttr>(valAttr);
+  assert(intAttr);
+  return intAttr.getInt();
+}
+
+bool fastPathAvailable(const SharedMemoryObject &smemObj, const SharedEncodingAttr &srcEncoding, const MfmaEncodingAttr &dstEncoding) {
+  if (dstEncoding.getNonKDim() != 32)
+    return false;
+  if (srcEncoding.getMaxPhase() > 1)
+    return false;
+  auto stride0 = findConstValue(smemObj.strides[0]);
+  auto stride1 = findConstValue(smemObj.strides[1]);
+  auto offset0 = findConstValue(smemObj.offsets[0]);
+  auto offset1 = findConstValue(smemObj.offsets[1]);
+  bool allValuesDefined =
+      stride0.has_value() &&
+      stride1.has_value() &&
+      offset0.has_value() &&
+      offset1.has_value();
+  if (!allValuesDefined)
+    return false;
+  if (offset0.value() != 0 || offset1.value() != 0)
+    return false;
+  return true;
+}
+
+// Computes offsets for operand A or transposed operand B
+// @param rewriter
+// @param loc
+// @param elemsPerInstr operand tile shape consumed by one MFMA instruction
+// @param waveM wave id for the "non K" axis
+// @param laneId lane id in warp [0..63]
+// @param warpsPerGroup number of warps in one block
+// @param numOfElems number of elements accessed by thread per repetition
+// @param reps number of instructions repretition to fully cover dot operand
+// @param cSwizzleOffset
+llvm::SmallVector<Value>
+fastPathComputeOffsetsTy1(ConversionPatternRewriter &rewriter, Location loc,
```

Comment:
`fastPathComputeOffsetsTy1` and `fastPathComputeOffsetsTy2` functions are taken from https://github.com/ROCmSoftwarePlatform/triton/pull/265/files#diff-fa6a8a176a627bd4941fbfe3555a3001107429c63d8119601f47736872587945R37